operators/src/opencl/cl_program.rs

```diff
@@ -2063,6 +2098,79 @@ __kernel void gid(
         assert_eq!(collection.multipoint_offsets(), &[0, 1, 3, 4]);
     }
 
+    #[test]
+    fn points_identity_with_time() {
+        let input = MultiPointCollection::from_data(
+            MultiPoint::many(vec![
+                vec![(0., 0.)],
+                vec![(1., 1.), (2., 2.)],
+                vec![(3., 3.)],
+            ])
+            .unwrap(),
+            vec![
+                TimeInterval::new_unchecked(0, 1),
+                TimeInterval::new_unchecked(1, 2),
+                TimeInterval::new_unchecked(2, 3),
+            ],
+            HashMap::new(),
+        )
+        .unwrap();
+        let typed_input = TypedFeatureCollection::MultiPoint(input.clone());
+
+        let mut out = FeatureCollection::<MultiPoint>::builder().batch_builder(3, 4);
+
+        let kernel = r#"
+            __kernel void points( 
+                __constant const double2 *IN_COLLECTION0_COORDS,
+                const int IN_COLLECTION0_COORDS_LEN,
+                __constant const int *IN_COLLECTION0_FEATURE_OFFSETS,
+                const int IN_COLLECTION0_FEATURE_OFFSETS_LEN,
+                __constant const long2 *IN_COLLECTION0_TIME,
+                const int IN_COLLECTION0_TIME_LEN,
+                __global double2 *OUT_COLLECTION0_COORDS,
+                const int OUT_COLLECTION0_COORDS_LEN,
+                __global int *OUT_COLLECTION0_FEATURE_OFFSETS,
+                const int OUT_COLLECTION0_FEATURE_OFFSETS_LEN,
+                __global long2 *OUT_COLLECTION0_TIME,
+                const int OUT_COLLECTION0_TIME_LEN
+            ) {
+                int idx = get_global_id(0);
+                
+                OUT_COLLECTION0_COORDS[idx] = IN_COLLECTION0_COORDS[idx];
+                
+                OUT_COLLECTION0_TIME[idx] = IN_COLLECTION0_TIME[idx];
+                
+                if (idx < IN_COLLECTION0_FEATURE_OFFSETS_LEN) {
+                    OUT_COLLECTION0_FEATURE_OFFSETS[idx] = IN_COLLECTION0_FEATURE_OFFSETS[idx];
+                }
+            }"#;
+
+        let mut cl_program = CLProgram::new(IterationType::VectorCoordinates);
+        cl_program.add_input_features(VectorArgument::new(
+            VectorDataType::MultiPoint,
+            vec![],
+            true,
+            true,
+        ));
+        cl_program.add_output_features(VectorArgument::new(
+            VectorDataType::MultiPoint,
+            vec![],
+            true,
+            true,
+        ));
+
+        let mut compiled = cl_program.compile(kernel, "points").unwrap();
+
+        let mut runnable = compiled.runnable();
+        runnable.set_input_features(0, &typed_input).unwrap();
+        runnable.set_output_features(0, &mut out).unwrap();
+        compiled.run(runnable).unwrap();
+
+        let result = out.output.unwrap().get_points().unwrap();
```

Comment:
please check if time is actually set correctly

Reply:
Here, I check if the result equals the input, i.e., it needs to have equal time intervals.